modules/api/src/main/java/org/apache/fluo/api/client/SnapshotBase.java

```diff
@@ -165,4 +167,22 @@
    * @return transactions start timestamp allocated from Oracle.
    */
   long getStartTimestamp();
+
+  default CompletableFuture<String> getsAsync(String row, Column column) {
+    return CompletableFuture.supplyAsync(() -> gets(row, column));
+  }
+
+  default CompletableFuture<String> getsAsync(String row, Column column, String defaultValue) {
+    return CompletableFuture.supplyAsync(() -> gets(row, column, defaultValue));
```

Comment:
Did you see my comment on the issue about supplyAsync?  These request will be put on a JVM wide thread pool and these request do I/O.  So they may block other unrelated work in the JVM, but not completely sure.  I will try to look into this some more.

Reply:
I did see but I thought it was OK for the default method to be blocking as long as the implementations weren't blocking. I can remove the default implementation all together.

Reply:
The [supplyAsync](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#supplyAsync-java.util.function.Supplier-) method will run the task on the ForkJoin common pool which is a JVM wide pool.  I was thinking the default method should just do something like the following to avoid this.

```java
  return CompletableFuture.completedFuture(gets(row, column, defaultValue));
```